compare_compression_algo.py

```diff
@@ -0,0 +1,62 @@
+## this code compares our compression algos
+from main import TensorStorage 
+import numpy as np
+from matplotlib import pyplot as plt
+import time
+from utils.dataset_utils import OriginalDataset
+
+def generate_comparison_table(performance_data):
+    table_header = "| Method | Compression Size (Kb) | Original Size (Kb) | Compression Speed (s) | Decompression Speed (s) | Compression Ratio |"
+    table_separator = "|--------|--------------------------|-----------------------|-----------------------|-------------------------|--------------------|"
+    table_rows = [
+        f"| {comp_algo} | {comp_size} | {orig_size} | {comp_time:.6f} | {decomp_time:.6f} | {comp_ratio:.2f} |"
+        for comp_algo, comp_size, orig_size, comp_time, decomp_time, comp_ratio in performance_data
+    ]
+
+    return "\n".join([table_header, table_separator] + table_rows)
+
+
+if __name__ == "__main__":
+    original_dataset_ = OriginalDataset(data_path="datasets/droid_100_sample_pictures")
+
+    comp_algo_list = ['FOR', 'delta']
+    performance_data = []
+    indx = 5 #index number of image we want to reconstruct
+
+    for comp_algo in comp_algo_list:
+        tensor_storage = TensorStorage(checkpoint=10,
+                                    original_dataset=original_dataset_,
+                                    encoding_scheme= comp_algo,
+                                    enlarge_factor=10)
+    # num_images = len(original_dataset_)
+        num_images = 100
+        print(f"#### Compressing  using {comp_algo} and storing {num_images} images #### ")
+        comp_start_time = time.time()
+        for idx in range(num_images):
+            tensor_storage.add()
+        comp_end_time = time.time()
+
+        total_comp_time = comp_end_time - comp_start_time
+        print(f'Compression Time of {comp_algo}: {total_comp_time} ')
+
+        decomp_start_time = time.time()
+        tensor_storage.get_image(indx)
+        decomp_end_time = time.time()
+
+        total_decomp_time = decomp_end_time - decomp_start_time
+        print(f'Decompression Time of {comp_algo}: {total_decomp_time} ')
+
+        comp_size = tensor_storage.get_size()
+        orig_size = original_dataset_.get_storage_size()
+
+        print(f"total size in kB of tensor storage: {comp_size}")
+        print(f"total size in kB of original dataset: {orig_size}")
+
+        comp_ratio = orig_size/comp_size
```

Comment:
also all the values can you correct them to the nearest 3sf, so that the final table alignment becomes right